lib/internal/queue.js

```diff
@@ -46,7 +47,11 @@ export default function queue(worker, concurrency, payload) {
                 q._tasks.push(item);
             }
         }
-        setImmediate(q.process);
+
+        if (!isWaitingForProcessing) {
+            isWaitingForProcessing = true;
+            setImmediate(q.process);
+        }
```

Comment:
Another way to implement this would be to add a  separate `scheduleProcessing` function that sets up `q.process` to run once on the next tick.  That way the logic isn't spread across `_insert` and `process`